source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -0,0 +1,513 @@
+		if(!foundMovementEnd && encoderPulsesInPeriod > 20){
+			qDebug() << "Found movement end at index " << x;
+			foundMovementEnd = true;
+		}
+		else if(foundMovementEnd && encoderPulsesInPeriod < 1){
+			foundMovementStart = true;
+			initiateMovementIndex = x;
+			qDebug() << "Found movement start at index " << initiateMovementIndex;
+		}
+	}
+
+
+	int scalerInitiateMovementIndex = 0;
+	bool foundScalerMovementStart = false;
+	bool foundScalerMovementEnd = false;
+
+	QVector<qint32> encoderUpVector = scalerChannelVectors.value("EncoderUp");
+	QVector<qint32> encoderDownVector = scalerChannelVectors.value("EncoderDown");
+
+	if(double(continuousConfiguration_->scanAxes().at(0)->regionAt(0)->regionStart()) < double(continuousConfiguration_->scanAxes().at(0)->regionAt(0)->regionEnd())){
+		for(int x = encoderUpVector.count()-1; (x > 0) && !foundScalerMovementStart; x--){
+			if(!foundScalerMovementEnd && encoderUpVector.at(x) > 20){
+				qDebug() << "Found scaler movement end at index " << x;
+				foundScalerMovementEnd = true;
+			}
+			else if(foundScalerMovementEnd && encoderUpVector.at(x) < 1){
+				foundScalerMovementStart = true;
+				scalerInitiateMovementIndex = x;
+				qDebug() << "Found scaler movement start index " << scalerInitiateMovementIndex;
+			}
+		}
+	}
+	else {
+		for(int x = encoderDownVector.count()-1; (x > 0) && !foundScalerMovementStart; x--){
+			if(!foundScalerMovementEnd && encoderDownVector.at(x) > 20){
+				qDebug() << "Found scaler movement end at index " << x;
+				foundScalerMovementEnd = true;
+			}
+			else if(foundScalerMovementEnd && encoderDownVector.at(x) < 1){
+				foundScalerMovementStart = true;
+				scalerInitiateMovementIndex = x;
+				qDebug() << "Found scaler movement start index " << scalerInitiateMovementIndex;
+			}
+		}
+	}
+
+	/* LEAVE THIS IN PLACE IN CASE WE HAVE TO GO BACK TO USING SCALER AS BASE TIME SCALE FOR EVERYTHING
+	for(int x = encoderUpVector.count()-6; (x > 5) && !foundScalerMovementStart; x--){
+
+		double runningAverage = (encoderUpVector.at(x+5)+encoderUpVector.at(x+4)+encoderUpVector.at(x+3)+encoderUpVector.at(x+2)+encoderUpVector.at(x+1)
+					 +encoderUpVector.at(x)+encoderUpVector.at(x-1)+encoderUpVector.at(x-2)+encoderUpVector.at(x-3)+encoderUpVector.at(x-4))/10;
+
+		if(!foundScalerMovementEnd && runningAverage > 1){
+			qDebug() << "Found scaler movement end at index " << x;
+			foundScalerMovementEnd = true;
+		}
+		else if(foundScalerMovementEnd && runningAverage < 0.1){
+			foundScalerMovementStart = true;
+			scalerInitiateMovementIndex = x;
+			qDebug() << "Found scaler movement start index " << scalerInitiateMovementIndex;
+		}
+	}
+	*/
+
+	qDebug() << "Amptek: " << initiateMovementIndex;
+	qDebug() << "Scaler: " << scalerInitiateMovementIndex;
+
+	if(!metaDataMap_.contains("GratingEncoderFeedback")){
+		qDebug() << "FATAL ERROR, MISSING READBACK VALUE FOR INITIAL ENCODER POSITION";
```

Comment:
How often do these fatal errors happen?  If it's ever possible via non-development means then there should be more cleanup code inside these if statements.